### PR TITLE
[CORE-2316] Add derived metadata to project defaults

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

**Review by bbonenfant — APPROVED** [review]

File: src/internal/ppsdb/ppsdb.proto [hunk 1]
```diff
@@ -3,12 +3,16 @@ syntax = "proto3";
...
+import "google/protobuf/timestamp.proto";
+
 message ClusterDefaultsWrapper {
   reserved 1, 2;
   reserved "details_json", "effective_details_json";
   string json = 3;
 }
 
 message ProjectDefaultsWrapper {
+  string created_by = 1;
```
Comment (acohen4) [inline src/internal/ppsdb/ppsdb.proto:15]: we used to have field 1 & 2 on this proto?

File: src/server/pps/server/api_server.go [hunk 2]
```diff
@@ -4130,7 +4134,21 @@ func (a *apiServer) SetProjectDefaults(ctx context.Context, req *pps.SetProjectD
...
+		// get old defaults in order to get created_by and created_at
+		projectDefaultsTxn := a.projectDefaults.ReadWrite(txnCtx.SqlTx)
+		// If auth is active, make caller the owner of these new project defaults.
+		var username string
+		if whoAmI, err := txnCtx.WhoAmI(); err == nil {
+			username = whoAmI.GetUsername()
+		} else if !errors.Is(err, auth.ErrNotActivated) {
+			return errors.Wrap(err, "could not get caller's username")
+		}
+		var new = &ppsdb.ProjectDefaultsWrapper{
```
Comment (acohen4) [inline src/server/pps/server/api_server.go:4146]: only weird bit of behavior is that the system may include project defaults that were created before this code and won't have created-by and created-at populated

**Review by acohen4 — APPROVED** [review]
Overall LGTM. Left a couple questions / considerations

File: src/internal/ppsdb/ppsdb.proto (hunk 1 above)
Comment (robert-uhl) [reply src/internal/ppsdb/ppsdb.proto:15]: Nope, `ProjectDefaultsWrapper` was a copy-paste of `ClusterDefaultsWrapper`, which _did_ have those fields, and maintains them as deprecated.

File: src/server/pps/server/api_server.go (hunk 2 above)
Comment (robert-uhl) [reply src/server/pps/server/api_server.go:4146]: That’s correct — they are not available, so are blank. We follow the same pattern for e.g. PFS projects.

**Pull request merged**